src/tools/compiletest/src/runtest.rs

```diff
@@ -4258,6 +4258,37 @@ impl<'test> TestCx<'test> {
                 V0_BACK_REF_RE.replace_all(&normalized, V0_BACK_REF_PLACEHOLDER).into_owned();
         }
 
+        // AllocId are numbered globally in a compilation session. This can lead to changes
+        // depending on the exact compilation flags and host architecture. Meanwhile, we want
+        // to keep them numbered, to see if the same id appears multiple times.
```

Comment:
@RalfJung what about this?
https://github.com/rust-lang/rust/pull/116810#discussion_r1361105504